src/BayesBaseH.java

```
@@ -728,7 +728,13 @@ public static void PropagateContextEdges() throws Exception {
          st_largest.close();
          
          Statement st1 = con2.createStatement();
-     	 ResultSet rs = st1.executeQuery("select name as RChain from lattice_set where lattice_set.length = 1 ;");
+        ResultSet rs = st1.executeQuery(
+            "select orig_rnid as RChain " +
+            "from lattice_set " +
+            "join lattice_mapping " +

```

Question: Do the lattice_set and lattice_mapping tables stay small?  One concern I have is that this join will become very expensive if the tables start to get large.

---

Hi Richard, about target_BN: this is more copy and past problems. The target table code solves some important problems, but we should separate it out from the structure learning. I wasn't planning to make it part of your thesis project. So for now, let's ignore the errors, and everything is working, we can move that code away.

---

About lattice_set join: this could be an issue. Indices may help. For now my main worry is that it takes too long to build those tables.

---

Okay, it sounds like we have the correct fix now :smile: